### DISCUSSION TITLE
Audio / IF Filter?

### DISCUSSION THREAD

**Discussion by ja-Bertolin** [AlexandreRouma/SDRPlusPlus#544] [General]
Hi, is there anyway to use an Audio / IF Filter with sdrpp? I was looking for some Open Source App that could help me to reduce noise background, improve audio using a Virtual Cable with sdrpp

Thanks

**Comment by corvus-ch** [comment]
Pulsaudio, which is the standard sound server on most Linux distros, can route the audio output of one application to the input of another application. Combine this with a host application that can rund audio pugins and you should be able to do exaclty that. As a starter, you can have a look at look at linux-audio.org.

Note, audio plugins can not run on their own, they need a host. Such a host could be Ardour. Ardour is a full fledged digital audio workstation. Total overkill for this case. There are more light weight options available. I can not make any reccomendations though as I do not have experience with any of them. A web search for "linux audio plugin host" or something along that line should bring um something useful.

While researching that topic, you will most likely stumble about the jack Audio Server. This is the a more capable alternative to Pulseaudio. As with Ardour, this could be used but is most likley overkill and not needed. Pulseaudio is more than capable for this scenario.

On macOS, Pulseaudio is not a thing. Hower it should be possible to get JackD to run. For those willing to spend some money, Loopback and Audio Hijack by Rogue Amoeba are greate tools. With Audio Hijack I was successfull in cleaning up some noise in my received signal.

Edit: Also cover macOS.

**Reply by ja-Bertolin** [reply]
Thanks, I will search more information about that. I am using MacOS, I hope I will be able to find something for it. SDR++ is becoming a very good alternative in MacOS to other SDR apps that just run in Windows, at least I like a lot, it has been done a great job with it. I only miss filters to reduce background noise. In any case, I am finishing the readiness of an external DSP(https://jackdev23.wixsite.com/iw2ndh-dsp)  that I will be able to connect to Audio output of the MacBook and will help in that way.

**Reply by corvus-ch** [reply]
Oh, my bad. I was asuming a linux machine. For MacOS this is a total different story. There Pulsaudio is not a thing. It should be possible to get JackD running though. If you want something that works (tm) out of the box, then have a look at the Loopback and Audio Hijack from Rogue Amoeba. With the later I have had success in exactly doing this.

**Comment by AlexandreRouma** [comment]
IF noise reduction for FM has been added (still a work in progress though) and audio noise reduction will be added soon

**Comment by ja-Bertolin** [comment]
Hi, when is it expected to deliver Audio Noise reduction? , any workarounds?
Thanks

**Comment by rogerxxxx** [comment]
Ditto?
